lib/handler/cache-handler.js

```diff
@@ -87,97 +99,109 @@ class CacheHandler {
+    }
 
-      const absoluteStaleAt = staleAt + baseTime
+    const resDate = typeof resHeaders.date === 'string'
+      ? parseHttpDate(resHeaders.date)
+      : undefined
+
+    const staleAt =
+      determineStaleAt(this.#cacheType, now, resAge, resHeaders, resDate, cacheControlDirectives) ??
+      this.#cacheByDefault
+    if (staleAt === undefined || (resAge && resAge > staleAt)) {
+      return downstreamOnHeaders()
+    }
+    const baseTime = resDate ? resDate.getTime() : now
+    const absoluteStaleAt = staleAt + baseTime
 
-      if (now >= absoluteStaleAt || (age && age >= staleAt)) {
-        // Response is already stale
+    if (now >= absoluteStaleAt) {
+      // Response is already stale
+      return downstreamOnHeaders()
+    }
+
+    let varyDirectives
+    if (this.#cacheKey.headers && resHeaders.vary) {
+      varyDirectives = parseVaryHeader(resHeaders.vary, this.#cacheKey.headers)
+      if (!varyDirectives) {
+        // Parse error
         return downstreamOnHeaders()
       }
+    }
 
-      let varyDirectives
-      if (this.#cacheKey.headers && headers.vary) {
-        varyDirectives = parseVaryHeader(headers.vary, this.#cacheKey.headers)
-        if (!varyDirectives) {
-          // Parse error
-          return downstreamOnHeaders()
-        }
-      }
+    const deleteAt = determineDeleteAt(baseTime, cacheControlDirectives, absoluteStaleAt)
+    const strippedHeaders = stripNecessaryHeaders(resHeaders, cacheControlDirectives)
+
+    /**
+     * @type {import('../../types/cache-interceptor.d.ts').default.CacheValue}
+     */
+    const value = {
+      statusCode,
+      statusMessage,
+      headers: strippedHeaders,
+      vary: varyDirectives,
+      cacheControlDirectives,
+      cachedAt: resAge ? now - resAge : now,
+      staleAt: absoluteStaleAt,
+      deleteAt
+    }
 
-      const deleteAt = determineDeleteAt(cacheControlDirectives, absoluteStaleAt)
-      const strippedHeaders = stripNecessaryHeaders(headers, cacheControlDirectives)
+    if (typeof resHeaders.etag === 'string' && isEtagUsable(resHeaders.etag)) {
+      value.etag = resHeaders.etag
+    }
 
-      /**
-       * @type {import('../../types/cache-interceptor.d.ts').default.CacheValue}
-       */
-      const value = {
-        statusCode,
-        statusMessage,
-        headers: strippedHeaders,
-        vary: varyDirectives,
-        cacheControlDirectives,
-        cachedAt: age ? now - (age * 1000) : now,
-        staleAt: absoluteStaleAt,
-        deleteAt
-      }
+    this.#writeStream = this.#store.createWriteStream(this.#cacheKey, value)
+    if (!this.#writeStream) {
+      return downstreamOnHeaders()
+    }
 
-      if (typeof headers.etag === 'string' && isEtagUsable(headers.etag)) {
-        value.etag = headers.etag
-      }
+    const handler = this
+    this.#writeStream
+      .on('drain', () => controller.resume())
+      .on('error', function () {
+        // TODO (fix): Make error somehow observable?
```

Comment:
If we cannot store the response's body, best will be to just handover the body to the caller and fail silently; I'm thinking on some sort of on `error` event here?

Reply:
> I'm thinking on some sort of on error event here?

Where would this be?

Reply:
Good question; maybe the `cache-handler` itself? I believe we do not have reference to the dispatcher at this point but the dispatch method.

Reply:
Let's explore it in further PR